First pull request on level-2 branch. Front-end part - responsive design layout (figma: dark-4)

letStayFoolish opened the pull request:
**Overview**:

In this pull request, I've meticulously implemented the frontend responsive design based on the provided design layout (dark-4) for the third phase of our thesis project. The entire frontend is crafted using React and adheres to JSX standards as required by our checklist.

I've gone above and beyond to ensure that every necessary element and component has been diligently integrated into the project. Leveraging the power of CSS BEM technology, I've meticulously styled the web application, creating an aesthetically pleasing and user-friendly interface.

The responsive design approach I've followed prioritizes desktop users, ensuring a seamless experience for them. While I've laid the foundation for logic and functions within the codebase, I've intentionally left the majority of this work for the next phase of development.

This pull request marks a significant milestone in our project, laying the groundwork for further enhancements and functionality. I look forward to your feedback and collaboration as we continue to refine and evolve this project.

letStayFoolish committed:
first commit

letStayFoolish committed:
Add: Navbar, Header (js + css)

letStayFoolish committed:
Add: Promo component (js + css)

letStayFoolish committed:
Add: Techs component (js + css)

letStayFoolish committed:
Update: Reposition between Header and Navbar, so Navbar would always be top over the Header component

letStayFoolish committed:
Update: Main Page done

Finish Components:
- JSX ok!
- CSS ok!
- Responsive ok!

letStayFoolish committed:
Update: Header, Navigation, Promo

Add Navigation component

Correction: Header(old) -> Promo(new) & Navbar(old) -> Header(new)

letStayFoolish committed:
Add: Burger-menu

Add animations
add burger-menu on screens < 769px

letStayFoolish committed:
Add: Movies & MoviesCard components

JSX + CSS

letStayFoolish committed:
Add: Movies, MoviesCard and SearchForm Responsive (CSS)

letStayFoolish committed:
Add: React Routing

Created routes for components:
- Main (home page)
- Movies
- Saved Movies

letStayFoolish committed:
Update: Compress all of PNG and SVG images

letStayFoolish committed:
Update: Navigation component

Account icon background color updates depending on path location (page URL)

letStayFoolish committed:
Add: NotFound Component

Created JSX and CSS + responsive component

letStayFoolish committed:
Add: Signup & Signin components

- JSX
- CSS3
- Responsive

letStayFoolish committed:
Add: Profile component

correct: Responsiveness
create: global components: Menu, Entry (signin + signup)

letStayFoolish committed:
Update: correction with toggleMenu. Added toggleMenu as a props to Menu component on small screens.

letStayFoolish committed:
Update: corrected Menu width on 768px and less

letStayFoolish committed:
Update: corrected SavedMovies component grid columns width on various device screens

letStayFoolish committed:
Update: Main page responsive corrections

letStayFoolish committed:
Update: NotFound component responsive corrections

letStayFoolish committed:
Update: Movies

Correction with media queries

letStayFoolish committed:
Update: Movies

Correction with media queries

letStayFoolish committed:
Update: Readme fil

letStayFoolish committed:
Update: Readme fil

letStayFoolish committed:
Update: Navigation component

Added overlay container on screen width 768px (only)

letStayFoolish committed:
Update: alt text changes in Menu.jsx

letStayFoolish commented:
**ДЛЯ** **РЕВЬЮЕРА**

Я попробовал сбилдить проект на разных пк, все работает, ошибок нет, есть только предупреждения. Кроме того проект успешно задеплоен на Vercel.

letStayFoolish committed:
Update: Input

Added placeholders on input fields.

nehydna commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
-  <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета. 

Хедер не должен быть липким http://joxi.ru/MAj5dwbiYPagG2, он должен скроллиться
На странице фильмов и сохраненных фильмов постеры должны быть в три колонки http://joxi.ru/Vm6YKX9IKa75Zr
Цвет фона попап не по макету (если открывать на главной) http://joxi.ru/52apXMvc07vRXm
Параметры кнопки закрытия не по макету http://joxi.ru/EA4YdOXI0Qazxr
Параметры шрифтов попап не по макету
На странице профайла надо поправить расположение элементов http://joxi.ru/n2YJL1ZC0vnWRA
На 768
постеры должны быть в две колонки http://joxi.ru/Vm6YKX9IKa79kr
На главной надо поправить расположение и размеры элементов http://joxi.ru/5md91EvsJy6lL2, http://joxi.ru/Q2KZg6LigxWk6m, http://joxi.ru/52apXMvc07vwXm, http://joxi.ru/823YW3zIwx5yyr, http://joxi.ru/8AnlawpcKPnMem
На странице профайла надо поправить расположение элементов http://joxi.ru/L21B5aXUDbLkbA
На 320 
Технологии должны быть в две колонки http://joxi.ru/p27YZxbIZ15Rwr
На страницах фильмов и сохраненных фильмов надо поправить расположение элементов http://joxi.ru/p27YZxbIZ15Jwr

- ✅ <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
-  <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить

Все ссылки на сторонние сайты должны открываться в новых вкладках. Например, http://joxi.ru/n2YJL1ZC0vnQPA
Текст на скрине http://joxi.ru/EA4YdOXI0Qa6Br  тоже надо сделать ссылками. То есть, одна ссылка должна оборачивать и текст, и картинку
Со страниц регистрации и авторизации не попасть на главную. Логотип надо сделать ссылкой

-  <!--5.55--> Именование классов сделаны по БЭМ

Необходимо использовать БЭМ-нотацию. В проекте есть одиночные модификаторы, элементы без блоков. Проверить БЭМ можно тут: https://yoksel.github.io/html-tree/


-  <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях

У всех элементов должен быть отступ снизу на десктопе и на всем адаптиве. Например, http://joxi.ru/DmBYgGzI6M8yMr. Это требование критерия **на всех промежуточных разрешениях**.  Отступ снизу не должен зависеть от высоты экрана, пользователю должно быть удобно управлять страницей на любой высоте, на любом разрешении.
На вем адаптиве верстка не должна ломаться. Например, http://joxi.ru/D2PngZ3tWVaEDr. Как вариант сделать более ранние точки перехода, либо более резиновую верстку.
Контент должен быть отцентрован http://joxi.ru/krDQgEjCdbBqRm, http://joxi.ru/VrwxywbCgPaQgm и т.д.. Надо исправить для всех похожих случаев

-  <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, обертки не состоят из див и т.д.)

Не должно быть ошибок в валидаторе. 
Большие разделы сайта должны быть секциями http://joxi.ru/v29YR1XIjoLnz2
Кнопки должны быть семантически размечены. Например, http://joxi.ru/gmvX1wbU0PZVdA. Надо исправить для всех похожих случаев
Лишних оберток в проекте быть не должно. Например, http://joxi.ru/KAxkpwbfwPRel2. Если див оборачиваеть только один тег, то этот див лишний

- ✅ <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- ✅ <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- ✅ <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- ✅ <!--3.38--> Работа модальных окон настроена.
- ✅ Разметка портирована в JSX:
	- ✅ <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- ✅ <!--3.38--> разметка вынесена в соответствующие ей компоненты

-  В проекте есть:
	- ✅ <!--3.38--> директория images/ с картинками,
            - ✅ <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- ✅ <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
-  <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.


Ссылки и кнопки должны менять прозрачность при наведении. Например, http://joxi.ru/RmzyBwbSVPOnKr, но надо исправить для всех интерактивных элементов
Кнопки выбранных фильмов должны выглядеть по макету http://joxi.ru/823YW3zIwx5Qnr

- ✅  <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- ✅  <!--2.14--> Шрифты подключены через @font-face.
- ✅ <!--2.14--> Иконки экспортированы в формате SVG.
- ✅ <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение
-  <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля

Поля не валидированы, плейсхолдеров нет

- ✅ <!--2.14--> reset.css запрещен.

### Рекомендации
-  <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.

Альтернативный шрифт не указан. Используйте font-family: 'Inter', ‘Arial’, sans-serif;

- ✅ <!--1.67--> Для изображений задан атрибут alt с подходящим значением.
- ✅ <!--1.67--> Растровые и векторные изображения оптимизированы;


## Количество баллов: 55
## Доп. Комментарии
- Если есть что-то, что не вошло в чеклист
Неверно указан атрибут lang

letStayFoolish commented:
Добрый день. Не понял ваш комментарий
 _На странице фильмов и сохраненных фильмов постеры должны быть в три колонки http://joxi.ru/Vm6YKX9IKa75Zr_
 У меня на 1280px картинки в 3 колонки как указано в ТЗ, дальше на 768 в 2 кол. и на 320 в 1 кол. 
 Screen: [1280px](https://drive.google.com/file/d/1oPPu7ie6HLu4f2_z-m0FF9KqOYMbsLqF/view?usp=drive_link)

letStayFoolish commented:
@nehydna На странице фильмов и сохраненных фильмов постеры должны быть в три колонки http://joxi.ru/Vm6YKX9IKa75Zr 
Добрый день. Не понял ваш комментарий
 _На странице фильмов и сохраненных фильмов постеры должны быть в три колонки http://joxi.ru/Vm6YKX9IKa75Zr_
 У меня на 1280px картинки в 3 колонки как указано в ТЗ, дальше на 768 в 2 кол. и на 320 в 1 кол. 
 Screen: [1280px](https://drive.google.com/file/d/1oPPu7ie6HLu4f2_z-m0FF9KqOYMbsLqF/view?usp=drive_link)

letStayFoolish commented:
@nehydna Также теперь у меня вопрос про Menu  на 768px и на 320px, как должен выгледить? Потому что на макете не указано -  "sticky(fixed)" или он должен скроллиться?

letStayFoolish commented:
@nehydna Уточните что необходимо сделать для валидации обязательных полей формы? в этом этапе мы не пишем функциональность, как я понял для валидации нужно делать запросы к бэку и тогда я смогу показывать ошибки при валидации. Пока я показал ошибку на инпуте с паролем, чтобы совпадало с макетом.

letStayFoolish committed:
Fix: Revision #1 (after first code review)

- Header: position: from fixed to absolute
- Navbar: menu background color change to dark on every page.
- Close button imported new icon and change positions on max-width: 768px, screen devices
- Menu: font size change to 18px on max-width: 768px
- Profile: because header needs to be !fixed added to profile__heading margin-top
- About Me: GitHub link - added opacity on hover & target="_blank"
- Project: list of project within <ul><li>...</li></ul>. h2 and img block wrapped within <a></a> link element.
- Signup & Signin: click on logo -> '/' page
- index.css: Add alternative font - Arial
- Fix: remove unnecessary <div> elements
- Update: Burger menu icon, exported from design
- Fix: Context centering horizontal
- Fix: Account div background
-

letStayFoolish committed:
Update: Clear code from unused elements

letStayFoolish commented:
@nehydna Почему пишете: "Цвет фона попап не по макету (если открывать на главной) http://joxi.ru/52apXMvc07vRXm" если в помакете указано раскрытое меню только на одной странице (/movies), а как должно выглядит на главной нигде не указано.?

letStayFoolish committed:
Update: Add required on Input elements

nehydna commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
-  <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета. 

На странице фильмов и сохраненных фильмов постеры должны быть в три колонки http://joxi.ru/4Ak6LY8cVwedb2, на 768 в две http://joxi.ru/YmE5gj9iLO6Nlm
Даю подсказку, так как ошибка редкая. В проекте для отступов используются относительные rem. Эта единица измерения зависит целиком от браузера, от его настроек. И если пользователь поставит свои настройки вместо тех, что браузер использует по умолчанию (причем, в любую сторону: в меньшую ли, в большую ли), то отображение изменится. Кроме того, не стоит забывать, что каждый браузер может ставить свои настройки и отображение может меняться также кроссбраузерно. Поэтому использовать ремы хорошо для шрифтов, но не для отступов. Так как у вас проект приближен к реальному, то необходимо учитывать поведение страниц при различных раскладах.

- ✅ <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- ✅ <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить
-  <!--5.55--> Именование классов сделаны по БЭМ

Необходимо использовать БЭМ-нотацию. Проверить БЭМ можно тут: https://yoksel.github.io/html-tree/


- ✅ <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях
-  <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, обертки не состоят из див и т.д.)

Не должно быть ошибок в валидаторе на всех страницах. Например, http://joxi.ru/4Ak6LY8cVwevb2. Валидными должны быть все страницы

- ✅ <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- ✅ <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- ✅ <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- ✅ <!--3.38--> Работа модальных окон настроена.
- ✅ Разметка портирована в JSX:
	- ✅ <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- ✅ <!--3.38--> разметка вынесена в соответствующие ей компоненты

-  В проекте есть:
	- ✅ <!--3.38--> директория images/ с картинками,
            - ✅ <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- ✅ <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
-  <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.

Ссылки и кнопки должны менять прозрачность при наведении. Например, http://joxi.ru/V2V76XduG5WbjA, но надо исправить для всех интерактивных элементов

- ✅  <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- ✅  <!--2.14--> Шрифты подключены через @font-face.
- ✅ <!--2.14--> Иконки экспортированы в формате SVG.
- ✅ <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение
-  <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля

Поля не валидированы 

- ✅ <!--2.14--> reset.css запрещен.

### Рекомендации
- ✅ <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- ✅ <!--1.67--> Для изображений задан атрибут alt с подходящим значением.
- ✅ <!--1.67--> Растровые и векторные изображения оптимизированы;


## Количество баллов: 75
## Доп. Комментарии
- Если есть что-то, что не вошло в чеклист

letStayFoolish committed:
Update: Convert rem to px

nehydna commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- ✅ <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета. 
- ✅ <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- ✅ <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить
-  <!--5.55--> Именование классов сделаны по БЭМ

Необходимо использовать БЭМ-нотацию на всех страницах. Проверить БЭМ можно тут: https://yoksel.github.io/html-tree/

- ✅ <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях
-  <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, обертки не состоят из див и т.д.)

Не должно быть ошибок в валидаторе. Например, на странице авторизации  http://joxi.ru/xAeoyjvhVk1dbm. Надо исправить для всех страниц проекта

- ✅ <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- ✅ <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- ✅ <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- ✅ <!--3.38--> Работа модальных окон настроена.
- ✅ Разметка портирована в JSX:
	- ✅ <!--3.38--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
	- ✅ <!--3.38--> разметка вынесена в соответствующие ей компоненты

-  В проекте есть:
	- ✅ <!--3.38--> директория images/ с картинками,
            - ✅ <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- ✅ <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- ✅ <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.

- ✅ <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- ✅  <!--2.14--> Шрифты подключены через @font-face.
- ✅ <!--2.14--> Иконки экспортированы в формате SVG.
- ✅ <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение
-  <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля

Поля не валидированы

- ✅ <!--2.14--> reset.css запрещен.

### Рекомендации
- ✅ <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- ✅ <!--1.67--> Для изображений задан атрибут alt с подходящим значением.
- ✅ <!--1.67--> Растровые и векторные изображения оптимизированы;


## Количество баллов: 89
## Доп. Комментарии
- Если есть что-то, что не вошло в чеклист
